src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -554,6 +554,17 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
                 hasErrors = true;
             }
 
+            if (node.IsKind(SyntaxKind.EqualsExpression) || node.IsKind(SyntaxKind.NotEqualsExpression))
+            {
+                if ((left.ConstantValue?.SpecialType == SpecialType.System_Double && left.ConstantValue.DoubleValue is double.NaN) ||
+                    (right.ConstantValue?.SpecialType == SpecialType.System_Double && right.ConstantValue.DoubleValue is double.NaN))
+                {
+                    var operatorText = node.IsKind(SyntaxKind.EqualsExpression) ? "==" : "!=";
```

Comment:
I'll consider not using magic strings here.

Reply:
It seems like `node.OperatorToken` would be good here?

Reply:
@RikkiGibson yes I think it should be the good approach here.
I'll get to that if it was decided to include this warning in the current warning wave (level 5).

It also seems that the test passes in CI, while it should fail because I haven't passed the expected diagnostics to `VerifyDiagnostics` yet. Not sure how to debug this issue if the call to `Error(diagnostics, ErrorCode.WRN_EqualityWithNaN, ....);` is hit.